draft-ietf-tls-tls13.md

```diff
@@ -4696,6 +4642,195 @@ a context in all exporter computations, though the value could be empty.
 Requirements for the format of exporter labels are defined in section 4
 of {{RFC5705}}.
 
+
+# 0-RTT and Anti-Replay {#anti-replay}
+
+As noted in {{zero-rtt-data}}, TLS does not provide inherent replay
+protections for 0-RTT data. There are two potential threats to be
+concerned with:
+
+- Network attackers who mount a replay attack by simply duplicating a
+  flight of 0-RTT data.  
+  
+- Network attackers who take advantage of client retry behavior
+  to arrange for the server to receive multiple copies of an application
+  message. This threat already exists
+  to some extent because browsers respond to network errors by
+  attempting to retry requests. However, 0-RTT adds an additional
+  dimension for any server system which does not maintain globally
+  consistent server state. Specifically, if a server system has
+  multiple zones where tickets from zone A will not be accepted in
+  zone B, then an attacker can duplicate a ClientHello and early
+  data intended for A to both A and B. At A, the data will
+  be accepted in 0-RTT, but at B the server will reject 0-RTT
+  data and instead force a full handshake. If the attacker, blocks
+  the ServerHello from A, then the client will complete the
+  handshake and retry the request, leading to duplication on
```

Comment:
"the client will complete the handshake will probably retry the request"; there's no firm guarantee when the connection is effectively restarted like this.